int8 id
float32[4] state
